packages/rest/src/validation/request-body.validator.ts

```diff
@@ -93,9 +92,11 @@ const compiledSchemaCache = new WeakMap();
 function validateValueAgainstJsonSchema(
   // tslint:disable-next-line:no-any
   body: any,
-  jsonSchema: object,
+  schema: object,
   globalSchemas?: SchemasObject,
 ) {
+  const jsonSchema = convertToJsonSchema(schema);
```

Comment:
Let's avoid possibly costly conversion from `schema` to `jsonSchema` in the case where we already have a cached validator.

```ts
function validateValueAgainstJsonSchema(...) {
  if (!compiledSchemaCache.has(schema)) {
    // convert schema to JSON schema
    // add global components
    // create AJV validator
    // store the validator in cache
  }

  if (validate(body)) {
     debug('Request body passed AJV validation.');
     return;
   }

  // handle validation errors
}
```

Ideally, the code building AJV validator should be extracted into a new function. E.g.

```ts
function validateValueAgainstJsonSchema(...) {
  let validate;
  if (compiledSchemaCache.has(schema)) {
    validate = compiledSchemaCache.get(schema);
  } else {
    validate = createValidator(/*...*/);
    compiledSchemaCache.set(schema, validator);
  }

  if (validate(body)) {
     debug('Request body passed AJV validation.');
     return;
  }

  // etc.
}
```